test/kvao/ttl.suite.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+var should = require('should');
+var helpers = require('./_helpers');
+var Promise = require('bluebird');
+
+module.exports = function(dataSourceFactory, connectorCapabilities) {
+  describe('ttl', function() {
+    var CacheItem;
+    beforeEach(function unpackContext() {
+      CacheItem = helpers.givenCacheItem(dataSourceFactory);
+    });
+
+    context('existing key with expire before expiration time', function() {
+      it('returns ttl - Callback API', function(done) {
+        CacheItem.set('a-key', 'a-value', 10, function(err) {
+          if (err) return done(err);
+          setTimeout(function() {
```

Comment:
I think the timeout is not needed here. It should not matter whether you are checking TTL 9 or 10 seconds before it expires. Unless I am missing something?